apps/prairielearn/src/ee/pages/instructorAiGenerateDrafts/instructorAiGenerateDrafts.ts

```diff
@@ -0,0 +1,78 @@
+import * as express from 'express';
+import asyncHandler from 'express-async-handler';
+
+import * as error from '@prairielearn/error';
+import { loadSqlEquiv, queryRows } from '@prairielearn/postgres';
+
+import { getCourseFilesClient } from '../../../lib/course-files-api.js';
+import { QuestionSchema } from '../../../lib/db-types.js';
+
+import {
+  InstructorAIGenerateDrafts,
+  draftMetadataWithQidSchema,
+} from './instructorAiGenerateDrafts.html.js';
+
+const router = express.Router();
+const sql = loadSqlEquiv(import.meta.url);
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_permission_edit) {
+      throw new error.HttpStatusError(403, 'Access denied (must be course editor)');
+    }
+
+    const drafts = (
+      await queryRows(
+        sql.select_draft_generation_info_by_course_id,
+        { course_id: res.locals.course.id },
+        draftMetadataWithQidSchema,
+      )
+    ).filter((x) => x.qid !== null);
+
+    res.send(InstructorAIGenerateDrafts({ resLocals: res.locals, drafts }));
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_permission_edit) {
+      throw new error.HttpStatusError(403, 'Access denied (must be course editor)');
+    }
+
+    if (req.body.__action === 'delete_drafts') {
+      const questions = await queryRows(
+        sql.select_drafts_by_course_id,
+        { course_id: res.locals.course.id.toString() },
+        QuestionSchema,
+      );
+
+      for (const question of questions) {
```

Comment:
we could also make a batch-delete editor and and remove all of the questions at once if that is preferred. would simply require modifying the trpc and making a new editor.

Reply:
I think a batch-delete would be much better here. A full delete can be a really heavy operation for larger courses. I'd be entirely happy to save the whole implementation of deletes for another PR.

Reply:
Deletion was indeed removed from this PR in f55db8a. We can add it back in the future.